packages/styled-system/src/utils/transform-functions.ts

```diff
@@ -82,3 +84,7 @@ export const transformFunctions = {
     return result
   },
 }
+
+const isCSSFunction = (value: unknown) => {
```

Comment:
Nice! I was thinking of a more complicated approach. i.e. regexing against possible bgImage gradients

linear-gradient
repeating-linear-gradient
radial-gradient
repeating-radial-gradient
conic-gradient
repeating-conic-gradient